rosbag2_compression/test/rosbag2_compression/test_zstd_compressor.cpp

```diff
@@ -22,32 +22,45 @@
 
 #include "gmock/gmock.h"
 
-class FilesystemHelperFixture : public rosbag2_test_common::TemporaryDirectoryFixture {};
 constexpr const char * GARBAGE_STATEMENT = "garbage";
 constexpr const int DEFAULT_GARBAGE_FILE_SIZE = 10 * 1024 * 1024;  // 10 MB
 constexpr const int DEFAULT_ITERATIONS = DEFAULT_GARBAGE_FILE_SIZE / sizeof(GARBAGE_STATEMENT);
 
-/**
- * Creates a 10 MiB file.
- * \param uri File path to write file.
- * \param num_iterations Number of times to write garbage statement.
- */
-void create_garbage_file(const std::string & uri, int num_iterations = DEFAULT_ITERATIONS)
+class CompressionHelperFixture : public rosbag2_test_common::TemporaryDirectoryFixture
 {
-  std::ofstream out{uri};
-  if (out) {
-    for (int i = 0; i < num_iterations; i++) {
-      out << GARBAGE_STATEMENT;
+protected:
+  CompressionHelperFixture() {}
+
+  /**
+   * Creates a 10 MiB file.
+   * \param uri File path to write file.
+   * \param num_iterations Number of times to write garbage statement.
+   */
+  void create_garbage_file(const std::string & uri, int num_iterations = DEFAULT_ITERATIONS)
+  {
+    std::ofstream out{uri};
+    if (out) {
+      for (int i = 0; i < num_iterations; i++) {
+        out << GARBAGE_STATEMENT;
+      }
+    } else {
+      throw std::runtime_error("Unable to write to file.");
```

Comment:
This could also be written as
```c++
if (!out) {
  throw std::runtime_error("Failed to write garbage file: " + uri);
}

for (int i = 0; i < num_iterations: i++) {
  out << GARBAGE_STATEMENT;
}
```